cli/inputflags/flags.go

```diff
@@ -66,3 +69,90 @@ func (f *Flags) Open(ctx context.Context, zctx *zed.Context, engine storage.Engi
 	}
 	return readers, nil
 }
+
+func (f *Flags) OpenWithStats(ctx context.Context, zctx *zed.Context, engine storage.Engine, paths []string, stopOnErr bool) ([]*StatsReader, error) {
+	var readers []*StatsReader
+	for _, path := range paths {
+		if path == "-" {
+			path = "stdio:stdin"
+		}
+		file, err := openFile(ctx, zctx, engine, path, f.ReaderOpts)
+		if err != nil {
+			err = fmt.Errorf("%s: %w", path, err)
+			if stopOnErr {
+				return nil, err
+			}
+			fmt.Fprintln(os.Stderr, err)
+			continue
+		}
+		readers = append(readers, file)
+	}
+	return readers, nil
+}
+
+func openFile(ctx context.Context, zctx *zed.Context, engine storage.Engine, path string, opts anyio.ReaderOpts) (*StatsReader, error) {
```

Comment:
It was kind of gross having to replicate the logic of anyio.Open here but I need to wrap the Readers to count the bytes read- didn't see another way. Open to suggestions.

Reply:
Create a storage.Engine wrapper with a Get method that, in turn, wraps a storage.Reader with something like your StatsReader before returning it and then provides a way to get stats for the wrapped readers!

Reply:
Great idea!